knotx-knot/knotx-knot-action/src/main/java/io/knotx/knot/action/ActionKnotProxyImpl.java

```diff
@@ -0,0 +1,259 @@
+/*
```

Comment:
I agree with @marcinczeczko 
Like the refactor very much but `ActionKnotProxyImpl` is too big and complicated. Please try to divide it into smaller logic units.